### PR TITLE
add validation to prevent contention between deployment and hpa controllers

### LINKED ISSUE #89: add validaton for the maxreplica field via a marker
size < than maxreplica and interger

### PR DESCRIPTION
**Pull request by skattoju**
fixes #89

### CODE REVIEW

File: internal/sub_reconcilers/reconcile_deployment_size.go [hunk 1]
```diff
@@ -60,6 +62,11 @@ func (r *DeploymentSizeReconciler) Reconcile(ctx context.Context, bestie *petsv1
 	if err != nil {
 		r.Log.Error(err, "unable to retrieve deployment")
 	}
+	// Validate that size <= MaxReplicas
+	if bestie.Spec.Size > *bestie.Spec.MaxReplicas {
```
Comment (acornett21) [inline internal/sub_reconcilers/reconcile_deployment_size.go:66]: This is confusing to me to read, the comment says `less than or equal to` but the code says `greater than`. Unless I don't know how to read which is quite possible. Maybe all english would better for the comment.

File: internal/sub_reconcilers/reconcile_hpa.go [hunk 2]
```diff
@@ -79,6 +81,11 @@ func (r *HPAReconciler) Reconcile(ctx context.Context, bestie *petsv1.Bestie) (c
 }
 
 func (r *HPAReconciler) createHorizontalPodAutoscaler(ctx context.Context, bestie petsv1.Bestie, bestieDeployment appsv1.Deployment) error {
+	//validate that MaxReplicas >= Size
+	if bestie.Spec.Size < *bestie.Spec.MaxReplicas {
```
Comment (acornett21) [inline internal/sub_reconcilers/reconcile_hpa.go:85]: Same comment as above.

**Review by acornett21 — COMMENTED** [review]
Just a few comments about the code comments.

File: internal/sub_reconcilers/reconcile_deployment_size.go (hunk 1 above)
Comment (yuriolisa) [reply internal/sub_reconcilers/reconcile_deployment_size.go:66]: @skattoju , Could you please rewrite the comment?

**Review by yuriolisa — CHANGES_REQUESTED** [review]

File: internal/sub_reconcilers/reconcile_deployment_size.go (hunk 1 above)
Comment (skattoju) [reply internal/sub_reconcilers/reconcile_deployment_size.go:66]: maths! but ok will un-reverse 😅

File: internal/sub_reconcilers/reconcile_hpa.go (hunk 2 above)
Comment (skattoju) [reply internal/sub_reconcilers/reconcile_hpa.go:85]: ok will un reverse brain

**Pull request merged**